src/coreclr/jit/llvm.cpp

```diff
@@ -1656,25 +1685,32 @@ void Llvm::storeLocalVar(GenTreeLclVar* lclVar)
 {
     if (lclVar->gtFlags & GTF_VAR_DEF)
     {
-        Value* valueRef = getGenTreeValue(lclVar->gtGetOp1());
-        assert(valueRef != nullptr);
+        Value* localValue = getGenTreeValue(lclVar->gtGetOp1());
+        assert(localValue != nullptr);
+
+        unsigned lclNum = lclVar->GetLclNum();
+        Value* allocaValue = (*m_allocas)[lclNum];
+        if (allocaValue != nullptr)
```

Comment:
I think it would be better to use `placeLocalInAlloca` here (maybe rename it to, say, `isLlvmFrameLocal`) instead of the `!= nullptr` test, for consistency.

Reply:
Done.